### PR TITLE
Refactor DB API to hide leaked abstractions

### PR DESCRIPTION
**Pull request by siy**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyTransaction.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package com.radixdlt.store.berkeley;
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyTransaction.java:1]: Missing license header

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyTransaction.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.radixdlt.store.Transaction;
+
+public class BerkeleyTransaction implements Transaction {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyTransaction.java:5]: Missing javadoc description + `final` keyword

Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyTransaction.java:5]: Also seems like this class can be package private

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/Transaction.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.store;
+
+public interface Transaction {
```
Comment (talekhinezh) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/Transaction.java:20]: Missing class description javadoc

**Review by talekhinezh — CHANGES_REQUESTED** [review]

**Review by clstrfsck — APPROVED** [review]

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**